File: php/lab2_php/make_operations_on_nrs.php
<?php

/**
 * Adds up numbers from input until zero is received. Prints sum.
 */
function printSumUntilZeroIsRead()
{
    $sum = 0;

    do {
        $nr = readline("Enter a number: ");
        $nr = (int) $nr;
        $sum += $nr;
    } while ($nr !== 0);

    echo "Sum is {$sum}";
}

// printSumUntilZeroIsRead();

/**
 * Prints divisors of number read fom input.
 */
function printDivisorsAndSumOfDivisors()
{
    $nr = readline("Enter a number: ");
    $nr = (int) $nr;

    $sum = 0;
    for ($i = 1; $i <= $nr; $i++) {
        if ($nr % $i == 0) {
            echo "Divisor {$i} was found\n";
            $sum += $i;
        }
    }

    echo "The sum of divisors is {$sum}";
}

// printDivisorsAndSumOfDivisors();

/**
 * Prints a count of the prime numbers given as input.
 */
function printPrimeCount()
{
    $n = readline("How many numbers? ");
    $n = (int) $n;

    $primeCount = 0;
    for ($i = 0; $i < $n; $i++) {
        $nr = readline("Enter a number: ");
        $nr = (int) $nr;

        if (isPrime($nr)) {
            $primeCount++;
        }
    }

    echo $primeCount;
}

/**
 * Silly && basic solution to identifying a prime number.
 *
 * @param int $nr
 * @return bool
 */
function isPrime(int $nr) : bool
{
    if ($nr <= 1) {
        return false;
    }

    for ($i = 2; $i < $nr; $i++) {
        if ($nr % $i == 0) {
            return false;
        }
    }

    return true;
}

function semiCleverIsPrime(int $nr)
{
    // only go halfway to $n
}

function moreThanSemiCleverIsPrime(int $nr)
{
    // only go up to sqrt($nr)
}

printPrimeCount();

File: py/lab2_py/make_operations_on_nrs.py
# Adds up numbers from input until zero is received. Prints sum.
def printSumUntilZeroIsRead():
	sum = 0;
	nr = None;

	while nr != 0:
		print("Enter a number: ");
		nr = input();
		nr = int(nr);
		sum += nr;

	print("Sum is: " + str(sum));

# printSumUntilZeroIsRead();

# Prints divisors of number read fom input.
def printDivisorsAndSumOfDivisors():
		print("Enter a number:");
		nr = input();
		nr = int(nr);

		sum = 0;
		i = 1;

		# freaking python for loops.
		# for i in [0..nr]:
		# use while so you don't just make a list for no reason at all.
		while i <= nr:
			if nr % i == 0:
				print("Divisor was found: " + str(i));
				sum += i;
			i += 1;

		print("The sum of divisors is: " + str(sum));

# printDivisorsAndSumOfDivisors();

# Prints a count of the prime numbers given as input.
def printPrimeCount():
	print("How many numbers?");
	n = input();
	n = int(n);

	primeCount = 0;
	i = 0;

	while i < n:
		print("Enter a number:");
		nr = input();
		nr = int(nr);
		
		if isPrime(nr):
			primeCount += 1; 
			i += 1;
	
	print(primeCount);

# Naive implementation for checking if a number is prime
def isPrime(nr):
	if nr <= 1:
		return false;

	i = 2;

	while (i < nr):
		# TODO condition and check

		i += 1;

	return true;